### DISCUSSION TITLE
Crypto operations in JavaScript functions within platform instruction limits

### DISCUSSION THREAD

**Discussion by vamshiaruru** [Shopify/function-examples#373] [Feedback: General Improvements]
Here's my usecase, each of my users have certain number of credits that are automatically applied as discounts during checkout flow if certain conditions are met. Since I can't fetch how many credits a user has from the function (since no api calls can be made), I was thinking I could put that data beforehand as a cart attribute in an encrypted format, decrypt it in my function and apply that as an order level discount.

But the documentation says 'crypto' is not available in functions. So, I am not quite sure how to decrypt on the function side anymore.

Any help is sincerely appreciated. Thank you.

**Comment by nickwesselman** [comment]
There are crypto libraries for JavaScript that do not depend on the browser `crypto`API. However in our testing, this will quickly hit the platform instruction count limit. This is a known limitation that we would like to resolve, but don't have a short term solution. If this is a hard requirement for you, I'd recommend using Rust.

**Reply by vamshiaruru** [reply]
thanks for the response! I shall explore rust.

**Comment by nickwesselman** [comment]
Renamed and moved to general improvements since this affects all APIs

**Comment by nboliver-ventureweb** [comment]
@nickwesselman  Any update on this feature? Our use case is similar: passing encrypted pricing data to a function via line item property.

**Reply by nickwesselman** [reply]
Not at the moment. Our recommendation at this time would still be to move to Rust, as is our general recommendation for functions which need to scale.

**Reply by nboliver-ventureweb** [reply]
Thanks, I appreciate the quick reply.
